### PR TITLE
Add gosimple

### LINKED ISSUE #1045: Run gosimple tool to as part of lint task
## Requirement - what kind of business use case are you trying to solve?

Enforce simple and readable code by running a simplicity checker (gosimple) as part of the lint job.

## Problem - what in Jaeger blocks you from solving the requirement?

gosimple is not used to check for unnecessarily complex code.

## Proposal - what do you suggest to solve the problem or improve the existing situation?

Use gosimple in lint task.

### PR DESCRIPTION
**Pull request by isaachier**
## Which problem is this PR solving?
- Resolves #1045.

## Short description of the changes
- Add gosimple to lint task. Apply existing suggestions to code.

### CODE REVIEW

File: cmd/ingester/main.go [hunk 1]
```diff
@@ -91,21 +91,19 @@ func main() {
 			consumer.Start()
 
 			hc.Ready()
-			select {
-			case <-signalsChannel:
-				logger.Info("Jaeger Ingester is starting to close")
-				err := consumer.Close()
+			<-signalsChannel
+			logger.Info("Jaeger Ingester is starting to close")
```
Comment (yurishkuro) [inline cmd/ingester/main.go]: nit: s/close/shutdown

Comment (isaachier) [reply cmd/ingester/main.go]: Sure. I can change the message too. Was just focusing on the issues gosimple pointed out.

Comment (isaachier) [reply cmd/ingester/main.go]: As a follow up, can we get rid of service name here? I don't see how this is more descriptive when the service logs are always kept separate. Even crossdock prefixes service names to logs.

Comment (yurishkuro) [reply cmd/ingester/main.go]: I thought the same but didn't want to look too pedantic. Yes the name is unnecessary.

**Pull request merged**